### ISSUE TITLE
Annotations event listener

### ISSUE THREAD

**Issue by Juanchoo055** [mozilla/pdf.js#16677]
Is there a way to add an event listener like "onchange" to the annotations when accessing them using let annotationData = await page.getAnnotations();? I have been attempting to do so, but haven't been successful.

**Comment by Snuffleupagus** [comment]
It's quite difficult to know exactly what you're asking about, but you might be able to achieve what you need by adding a `onSetModified` callback-function to the `annotationStorage`-instance since that's invoked every time form-data changes; see https://github.com/mozilla/pdf.js/blob/73d650af3e88037c03f89aecdadf4824ac99f2a7/src/display/annotation_storage.js#L35-L39

---

Unfortunately it's very difficult to help here since it seems that you've focused more on a particular solution, rather than describing your actual "problem" in sufficient detail.
When opening an issue, please provide *all* of the information requested in https://github.com/mozilla/pdf.js/blob/master/.github/ISSUE_TEMPLATE.md including a clear and concise description of the problem.

**Comment by Juanchoo055** [comment]
I am sorry yo didnt understand wath i am asking, i will say it  different way,

I am sorry for any confusion. Let me rephrase it for clarity.

I have a PDF file that contains input fields. After some research, I discovered that I can access these input fields using the page.getAnnotations() function. This function provides an object that contains all the input fields, including their types and content.

I have successfully implemented a feature that allows users to interact with the PDF by displaying the input annotations, enabling them to write on the fields, as you can see next code.

`async setupAnnotationss(page, viewport) {
      let annotationData = await page.getAnnotations();

      const pdf_canvas = document.getElementById('pdf-canvas');
      // Canvas offset
      const canvas_offset = pdf_canvas.getBoundingClientRect();
      // Canvas height
      const canvas_height = pdf_canvas.height;
      // Canvas width
      const canvas_width = pdf_canvas.width;
      // CSS for annotation layer
      const annotationLayer = document.getElementById('annotation-layer');
      annotationLayer.style.left = canvas_offset.left + 'px';
      annotationLayer.style.top = canvas_offset.top + 'px';
      annotationLayer.style.height = canvas_height + 'px';
      annotationLayer.style.width = canvas_width + 'px';

      // //Renderizar las nuevas anotaciones
      pdfjsLib.AnnotationLayer.render({
        viewport: viewport.clone({ dontFlip: true }),
        div: annotationLayer,
        annotations: annotationData,
        page: page,
        event: onclick,
      });
    },`

However, I'm currently working on adding an event listener to each annotation. The purpose of this event listener is to identify which values are being changed by the user. Furthermore, I aim to provide the users with additional options based on these changes, enhancing their experience with the PDF.
